### DISCUSSION TITLE
RDkit 2023.03.2: is there a way to GetMolSize() correctly after specifying the fixedBondLength?

### DISCUSSION THREAD

**Discussion by pavelVPo** [rdkit/rdkit#6562] [Q&A] [answered]
Hello,
I need to get the size of the molecule to draw it with fixedBondLength option turned on. The problem is that GetMolSize() seems to ignore this option: 
>>> from rdkit import Chem
>>> from rdkit.Chem.Draw import rdMolDraw2D
>>> mol = Chem.MolFromSmiles('C=CCC1(CC(C)O)C(=O)NC(=O)NC1=O')
>>> sizer = rdMolDraw2D.MolDraw2DSVG(-1,-1)
>>> sizer.drawOptions().fixedBondLength = **15**
>>> size = sizer.GetMolSize(mol)
>>> size
**(197, 142)**
>>> mol = Chem.MolFromSmiles('C=CCC1(CC(C)O)C(=O)NC(=O)NC1=O')
>>> sizer = rdMolDraw2D.MolDraw2DSVG(-1,-1)
>>> sizer.drawOptions().fixedBondLength = **500**
>>> size = sizer.GetMolSize(mol)
>>> size
**(197, 142)**

Am I doing something wrong? Is there RDkit way to compute the molecule’s size considering the fixedBondLength option?
Thank you in advance.

**Comment by ptosco** [comment, marked as answer]
Based on [what I can gather from the documentation](https://www.rdkit.org/docs/source/rdkit.Chem.Draw.rdMolDraw2D.html#rdkit.Chem.Draw.rdMolDraw2D.MolDraw2D.SetFlexiMode), it looks like `flexicanvas` (that you choose when you set the `MolDraw2D` size to `(-1, -1)` is mutually exclusive with `fixedBondLength` and `fixedScale`.
However, you can leverage [`scalingFactor` (defaults to 20)](https://www.rdkit.org/docs/source/rdkit.Chem.Draw.rdMolDraw2D.html?highlight=moldrawoptions#rdkit.Chem.Draw.rdMolDraw2D.MolDrawOptions.scalingFactor):
```
mol = Chem.MolFromSmiles('C=CCC1(CC(C)O)C(=O)NC(=O)NC1=O')
sizer = rdMolDraw2D.MolDraw2DSVG(-1, -1)
sizer.GetMolSize(mol)
(197, 142)

sizer = rdMolDraw2D.MolDraw2DSVG(-1, -1)
sizer.drawOptions().scalingFactor = 10
sizer.GetMolSize(mol)
(98, 71)
```

**Comment by pavelVPo** [comment]
Thank you for pointing this out. It works for me.
